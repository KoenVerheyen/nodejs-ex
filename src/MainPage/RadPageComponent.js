// @flow
import React from 'react'
import getBreadCrumb from './BreadCrumb'
import type {Match, Location} from "react-router-dom";
import RadRoutingComponent from "./RadRoutingComponent";

type Props = {
    match: Match,
    location: Location
}

export function RadPageComponent({match, location} : Props) {
    return (
        <div
            title='RAD'
            breadcrumb={getBreadCrumb(location)}>
            <RadRoutingComponent match={match} location={location}/>
        </div>)
}

export default RadPageComponent