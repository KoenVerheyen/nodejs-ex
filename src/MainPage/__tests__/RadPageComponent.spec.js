import React from 'react'
import RadPageComponent from '../RadPageComponent'
import {shallow} from 'enzyme'



describe('RadPageComponent', () => {

    test('Standard', () => {
        let wrapper = shallow(<RadPageComponent match={{path: '/rad'}} location={{pathname: '/rad/query'}}/>)

        expect(wrapper).toMatchSnapshot()
        expect(wrapper.length).toEqual(1)
    })

})