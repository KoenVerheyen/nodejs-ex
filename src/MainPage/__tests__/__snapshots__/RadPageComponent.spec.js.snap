// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadPageComponent Standard 1`] = `
<div
  breadcrumb={
    Array [
      Object {
        "link": "/rad",
        "title": "Rad Home",
      },
      Object {
        "link": "/rad/project",
        "title": "AIRAC amendments",
      },
      Object {
        "link": "/rad/rs",
        "title": "Restrictions",
      },
      Object {
        "link": "/rad/az",
        "title": "Aerodrome sets",
      },
    ]
  }
  title="RAD"
>
  <RadRoutingComponent
    location={
      Object {
        "pathname": "/rad/query",
      }
    }
    match={
      Object {
        "path": "/rad",
      }
    }
  />
</div>
`;
